### DISCUSSION TITLE
Cyclops Level Builder as a Separate Application (not a godot addon)

### DISCUSSION THREAD

**Discussion by Gamepro5** [blackears/cyclopsLevelBuilder#161] [General]
I was wanted to use cyclops in a level builder app I'm working on that is separate from the Godot editor. You mentioned before that you were considering making a branch of the level builder that doesn't run on the editor because there are a lot of gizmo problems. I'm not the author of this repo so I don't really even know how one would go about packaging this in a Godot project and not an addon, but I'd love to learn how it could happen and help.

**Comment by blackears** [comment]
The first big step is to get Cyclops working in a separate window and not the editor viewport.  Once that is done, it ought to be a lot easier to have it run as a stand alone application.  I need to do a lot of rewriting to untangle things first, though.  The main reason I want to separate things is go get more control over the viewport and also hopefully fix a lot of stability issues.  GDScipt seems to have a lot of problems and is not as reliable as a typical compiled language.

**Reply by Gamepro5** [reply]
Is this something you want to do or are actively working on?

**Reply by blackears** [reply]
I'm actively working on it now.  A focus in 1.0.3 was preparing things to make this move.

**Reply by Gamepro5** [reply]
Sounds good. Let me know if I can do anything to help.

**Reply by blackears** [reply]
I don't think so. This is going to be a tricky redesign and is more of a one person job.

**Reply by yessfan** [reply]
One thing I saw with the Castagne fighting game engine (https://castagneengine.com/) is that it is just Godot but it already has a scene that when run will open up the fighting game editor. You can also open up the game you are making using a different option in the editor menu. This I feel could be a very intuitive and powerful method, since Godot already has strong application building tools. You can even use a strong, consistent systems language if you care for C++ using that new GDExtention stuff, and if you don't like C++ and can work around the potential issues then you could probably also use one of the other languages with GDE bindings like Rust.

**Comment by MachinistJohn** [comment]
Any update on this? Does it live or is it dead? No pressure just asking, lol.

**Comment by blackears** [comment]
I have no plans to do this any time soon.  This would rely on a lot of other things being changed first, which will likely take several months at my current rate.
